modules/gmc-agent/src/main/java/org/gridgain/service/tracing/GmcSpanExporter.java

```diff
@@ -111,25 +84,25 @@ public void sendInitialState() {
      */
     TimeLimitedHandler getTraceHandler() {
         return new TimeLimitedHandler(Tracing.getTracer(), Duration.create(10, 0), "SendGmcSpans") {
-            /** Buffer. */
-            private final List<Span> buf = Collections.synchronizedList(new ArrayList<>());
-
-            @Override public void timeLimitedExport(Collection<SpanData> spanDataList) throws Exception {
-                spanDataList.forEach(s -> buf.add(fromSpanDataToSpan(s)));
+            @Override public void timeLimitedExport(Collection<SpanData> spanDataList) {
+                List<Span> spans = spanDataList.stream()
+                        .map(GmcSpanExporter::fromSpanDataToSpan)
+                        .collect(Collectors.toList());
 
-                if (log.isDebugEnabled())
-                    buf.forEach((s) -> log.debug("Sending span to GMC: " + s));
+                ClusterNode node = ctx.grid().cluster().forServers().nodes().stream()
+                        .min(Comparator.comparingLong(ClusterNode::order))
+                        .orElse(null);
 
-                if (mgr.send(buildSaveSpanDest(ctx.cluster().get().id()), Lists.newArrayList(buf)))
-                    buf.clear();
+                IgniteMessaging msg = ctx.grid().message(ctx.grid().cluster().forNode(node));
```

Comment:
Why not org.apache.ignite.cluster.ClusterGroup#forOldest?

Reply:
Thanks :)